plugin-server/src/main/ingestion-queues/session-recording/session-recordings-blob-consumer.ts

```diff
@@ -97,11 +97,11 @@ export class SessionRecordingBlobIngester {
                 session_id,
                 partition,
                 topic,
-                (offsets) => {
+                (offsets, self: SessionManager) => {
                     this.offsetManager?.removeOffsets(topic, partition, offsets)
 
                     // If the SessionManager is done (flushed and with no more queued events) then we remove it to free up memory
-                    if (sessionManager.isEmpty) {
+                    if (self.isEmpty) {
                         this.sessions.delete(key)
                     }
```

Comment:
Alternative: Why don't we clean this up in the periodic flush checker? It's only an optimisation to make sure we don't hold a load of empty objects in memory

Reply:
Way nicer 👨‍🍳 👌